webapp/src/js/services/enketo.js

```diff
@@ -530,6 +532,54 @@ angular.module('inboxServices').service('Enketo',
       form.output.update();
     };
 
+    // REVIEW this is quite big and unrelated to enketo itself; maybe it should
+    // be a separate service?  Or combined into the current Form2sms service as
+    // e.g. FormsAsSms service?
+    function submitFormBySmsIfApplicable(doc) {
```

Comment:
Agree, this should be a separate service. You have:
 - Enketo, which handles bootstrapping and parsing results from that beast
 - This, which handles extra SMS sending logic
 - FormsAsSms, which handles the api understanding and parsing of how to convert a form into an SMS